test/app_test.dart

```diff
@@ -62,11 +62,89 @@ Future<void> main([List<String>? args]) async {
     final app = App(configuration);
     final credentials = Credentials.anonymous();
     final user = await app.logIn(credentials);
+    expect(user.state, UserState.loggedIn);
   });
+
   test('Application get all users', () {
     final configuration = AppConfiguration(generateRandomString(10));
     final app = App(configuration);
     var users = app.users;
     expect(users.isEmpty, true);
   });
+
+  baasTest('App log out no current user is no operation and does not crash', (configuration) async {
+    final app = App(configuration);
+    await app.logout();
+  });
+
+  baasTest('App log out user', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+
+    expect(user.state, UserState.loggedIn);
+    await app.logout(user);
+    expect(user.state, UserState.loggedOut);
+  });
+
+  baasTest('App remove user', (configuration) async {
+    final app = App(configuration);
+    final user = await app.logIn(Credentials.emailPassword(testUsername, testPassword));
+
+    expect(user.state, UserState.loggedIn);
+    await app.removeUser(user);
+    expect(user.state, UserState.removed);
+  });
+
+  baasTest('App log out anon user is marked as removed', (configuration) async {
```

Comment:
```suggestion
  baasTest('App log out anonymous user is marked as removed', (configuration) async {
```

Reply:
Do we really need to spell this in full. I had this written like that but then decided against it